content/posts/event-driven-scaling-fission-function-using-keda.md

```diff
@@ -0,0 +1,239 @@
+---
+title: "Event Driven Scaling Fission Function Using KEDA"
+date: 2020-07-14T18:09:02+05:30
+author: "Fission"
+categories: ["Fission", "Serverless", "go", "Keda", "Event-driven", "Apache Kafka", "Tutorial", "Faas"]
+description: "Enabling event driven autoscaling with KEDA integration"
+type: "post"
+---
+
+# Introduction
+
+Events and integrations with event sources such as message queues are an important part of running functions. Fission had the MQ integration for invoking functions and this integration was available for Kafka, NATS, and Azure Queue Storage. This integration had a few limitations:
+
+- For every new integration you want to enable - there was one pod running for enabling the integration
+- Autoscaling of the trigger handler was not available.
+- Lastly - one Fission installation could connect to and handle only one instance of a MQ. So for example you could only connect to one Kafka instance in a given Fission installation. 
+
+When we decided to build out these features we came across Keda project and it solved most of the limitations which Fission had in the event integration area. Let’s dive into how it works!
+
+# Architecture
+
+Let’s first understand how the whole thing works. We won’t explain Keda in detail but you can take a look at Keda documentation for details.
+
+
+![architecture](../../../images/event-driven-scaling-fission-function-using-keda/architecture.png)
```

Comment:
This format will not work - please check other image samples

Reply:
Done